optimum/exporters/openvino/model_patcher.py

```diff
@@ -288,7 +289,82 @@ def __exit__(self, exc_type, exc_value, traceback):
             block.self_attention.core_attention.forward = block.self_attention.core_attention._orig_forward
 
 
+# transformers/models/llama/modeling_llama.py
+# transformers/models/gemma/modeling_gemma.py
+def _update_causal_mask(self, attention_mask, input_tensor, cache_position):
+    if self.config._attn_implementation == "flash_attention_2":
+        if attention_mask is not None and 0.0 in attention_mask:
+            return attention_mask
+        return None
+
+    dtype, device = input_tensor.dtype, input_tensor.device
+    min_dtype = torch.finfo(dtype).min
+
+    if dtype == torch.float32 or dtype == torch.float:
+        min_dtype = torch.finfo(torch.bfloat16).min
```

Comment:
what do you think about using torch.finfo(torch.float16) instead? We tested yesterday and sow that it also works without accuracy issue. gemma has 2b variant that can be fit to arm cpu with fp16, I think it also can get the same overflow on this platform (just for avoiding fixing that twice)

Reply:
Good point, fp16 should be better!